### PR TITLE
Test fixes for CR-GCE

### PR DESCRIPTION
**Pull request by adarshdubey-star**
**Test results**

```
pytest  tests/foreman/api/test_computeresource_gce.py::TestGCEHostProvisioningTestCase
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.8.6, pytest-7.1.1, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/addubey/work/robottelo, configfile: pyproject.toml
plugins: reportportal-5.0.12, services-2.2.1, mock-3.7.0, forked-1.3.0, ibutsu-2.0.2, cov-3.0.0, xdist-2.5.0
collected 3 items                                                                                                                                                                                                 

tests/foreman/api/test_computeresource_gce.py ...                                                                                                                                                           [100%]
=================================================================================== 3 passed, 3 warnings in 3020.45s (0:50:20) ====================================================================================
```

### CODE REVIEW

**Comment by shweta83** [comment]
trigger: robottelo
pytest: tests/foreman/api/test_computeresource_gce.py

File: pytest_fixtures/component/provision_gce.py [hunk 1]
```diff
@@ -26,6 +27,21 @@ def gce_cert(default_sat):
...
+@pytest.fixture(scope='session')
+def gce_cert_puppet(session_puppet_enabled_sat):
+    _, gce_cert_file = mkstemp(suffix='.json')
+    cert = json.loads(settings.gce.cert)
+    cert['local_path'] = gce_cert_file
+    with open(gce_cert_file, 'w') as f:
+        json.dump(cert, f)
+    session_puppet_enabled_sat.put(gce_cert_file, settings.gce.cert_path)
+    if session_puppet_enabled_sat.execute(f'[ -f {settings.gce.cert_path} ]').status != 0:
+        raise GCECertNotFoundError(
```
Comment (mshriver) [inline pytest_fixtures/component/provision_gce.py:39]: I think it would be better to call `pytest.fail` here to explicitly fail the setup.

**Review by mshriver — APPROVED** [review]

File: tests/foreman/api/test_computeresource_gce.py [hunk 2]
```diff
@@ -65,6 +65,26 @@ def module_gce_finishimg(
     return finish_image
 
 
+@pytest.fixture(scope='module')
+def module_gce_finishimg_puppet(
```
Comment (pondrejk) [inline tests/foreman/api/test_computeresource_gce.py:69]: for a second I thought it should be `finishing`, maybe `finish_img` would read nicer?

**Review by pondrejk — APPROVED** [review]
ack, minor comment

File: tests/foreman/api/test_computeresource_gce.py (hunk 2 above)
Comment (adarshdubey-star) [reply tests/foreman/api/test_computeresource_gce.py:69]: Thanks for your input @pondrejk I will keep this in mind for upcoming PR's and replicate universally.

**Pull request merged**